adapters/multiply.go

```diff
@@ -9,9 +9,28 @@ import (
 	"github.com/smartcontractkit/chainlink/store/models"
 )
 
+// Multiplier represents the number to multiply by in Multiply adapter.
+type Multiplier float64
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (m *Multiplier) UnmarshalJSON(input []byte) error {
+	if isString(input) {
+		input = input[1 : len(input)-1]
+	}
```

Comment:
There might be a better way to extract the string from the quotes with Marshal rather than resorting to manual indexing:

```golang
package main

import (
	"fmt"
	"encoding/json"
)

func main() {
	bytes, _ := json.Marshal("somevalue")
	fmt.Println("Hello, playground", string(bytes))
	
	var dem string
	json.Unmarshal(bytes, &dem)
	fmt.Println("yooo", dem)
}
```

```
Hello, playground "somevalue"
yooo somevalue
```

Give it a shot.

Reply:
It cannot unmarshal between different types: https://play.golang.org/p/rLruSyc3jsn `input` can be a number (`[49 50 51]`, 123 as bytes) or a string (`[34 49 50 51 34]`, "123" as bytes). When `dem` is `string`, number cannot be unmarshalled; when it's `int/float64`, strings cannot be unmarshelled.